src/v/cluster/partition.cc

```diff
@@ -730,20 +730,40 @@ ss::future<> partition::update_configuration(topic_properties properties) {
     // Before applying change, consider whether it changes cloud storage
     // mode
     bool cloud_storage_changed = false;
+
+    bool old_archival = old_ntp_config.is_archival_enabled();
     bool new_archival = new_ntp_config.shadow_indexing_mode
                         && model::is_archival_enabled(
                           new_ntp_config.shadow_indexing_mode.value());
 
+    bool old_read_replica = old_ntp_config.is_read_replica_mode_enabled();
+    bool new_read_replica = new_ntp_config.read_replica.value_or(false);
+
     bool old_compaction_status = old_ntp_config.has_compacted_override();
     bool new_compaction_status
       = new_ntp_config.cleanup_policy_bitflags.has_value()
         && model::is_compaction_enabled(
           new_ntp_config.cleanup_policy_bitflags.value());
-    if (
-      old_ntp_config.is_archival_enabled() != new_archival
-      || old_ntp_config.is_read_replica_mode_enabled()
-           != new_ntp_config.read_replica.value_or(false)
-      || old_compaction_status != new_compaction_status) {
+
+    if (old_archival != new_archival) {
+        vlog(
+          clusterlog.debug,
+          "[{}] updating archiver for topic config change in "
+          "archival_enabled",
+          _raft->ntp());
+        cloud_storage_changed = true;
+    } else if (old_read_replica != new_read_replica) {
+        vlog(
+          clusterlog.debug,
+          "[{}] updating archiver for topic config change in "
+          "read_replica",
+          _raft->ntp());
+        cloud_storage_changed = true;
+    } else if (old_compaction_status != new_compaction_status) {
```

Comment:
It could be just an `if`. `else if` ensures that we only log once for restarts and we log the first reason that caused a restart. It's possible to have multiple config changes in a single topic update that cause a restart, so I could make it an `if` to provide more context.

Reply:
Fixed